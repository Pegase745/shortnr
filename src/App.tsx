import { RouteComponentProps, Router } from '@reach/router';
import * as React from 'react';
import Message from 'react-bulma-components/lib/components/message';

import Home from './containers/Home';
import Redir from './containers/Redir';

export const NotFound: React.SFC<RouteComponentProps> = () => (
  <Message color="danger">
    <Message.Body>
      <h4>Whoops</h4>
      <p>Sorry, nothing to see here.</p>
    </Message.Body>
  </Message>
);

const App = () => {
  return (
    <Router>
      <Home path="/" />
      <Redir path="/r/:shortURL" />
      <NotFound default={true} />
    </Router>
  );
};

export default App;
