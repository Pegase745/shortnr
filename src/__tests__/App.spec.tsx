import { shallow } from 'enzyme';
import * as React from 'react';
import Message from 'react-bulma-components/lib/components/message';

import App, { NotFound } from '../App';

describe('App.tsx', () => {
  it('should render', () => {
    const app = shallow(<App />);

    expect(app).toMatchSnapshot();
  });

  it('should render NotFound', () => {
    const wrapper = shallow(<NotFound />, { disableLifecycleMethods: true });

    expect(wrapper.find(Message)).toExist();
  });
});
