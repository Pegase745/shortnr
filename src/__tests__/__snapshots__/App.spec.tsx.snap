// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.tsx should render 1`] = `
<Router>
  <Home
    path="/"
  />
  <Redir
    path="/r/:shortURL"
  />
  <Component
    default={true}
  />
</Router>
`;
